### ISSUE TITLE
[oracle] Numbers like 1. can't be parsed

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#8110]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting the example below leads to a 'Found unparsable section' error.

### Expected Behaviour

The example should work fine.

### Observed Behaviour

Linting the example below gives the following errors:
`Line 3, Position 14: Found unparsable section: '= 1.;'`
`Unable to lex characters: '1.;'`

### How to reproduce

```sql
select ex_col
from ex_tab
where ex_col = 1.;
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
